### PR TITLE
[SPARK-34863][SQL][FOLLOWUP] Add benchmark for Parquet & ORC nested column scan

### PR DESCRIPTION
**Pull request by sunchao**
### What changes were proposed in this pull request?

This adds benchmark for Parquet & ORC nested column scan, e.g., struct, list and map.

### Why are the changes needed?

Both Parquet and ORC now support vectorized reader for nested column now, but there is no benchmark to measure the performance yet.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

N/A - benchmark only.

### CODE REVIEW

**Comment by sunchao** [comment]
cc @dongjoon-hyun @viirya @LuciferYang @cloud-fan @c21 could you review this? thanks!

**Review by viirya — APPROVED** [review]

**Review by viirya — COMMENTED** [review]
Is the benchmark result performed by GA?

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
> Is the benchmark result performed by GA?

Yes, it looks like that, @viirya .

File: sql/core/benchmarks/DataSourceReadBenchmark-jdk11-results.txt [hunk 1]
```diff
@@ -2,322 +2,430 @@
...
+
+OpenJDK 64-Bit Server VM 11.0.14+9-LTS on Linux 5.13.0-1021-azure
+Intel(R) Xeon(R) Platinum 8171M CPU @ 2.60GHz
+SQL Single INT Column Scan in Struct:                        Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
+-------------------------------------------------------------------------------------------------------------------------------------------
+SQL ORC MR                                                            2376           2380           6          6.6         151.0       1.0X
+SQL ORC Vectorized (Nested Column Disabled)                           2333           2378          64          6.7         148.4       1.0X
+SQL ORC Vectorized (Nested Column Enabled)                             430            451          20         36.6          27.3       5.5X
+SQL Parquet MR: DataPageV1                                            2485           2501          22          6.3         158.0       1.0X
+SQL Parquet Vectorized: DataPageV1 (Nested Column Disabled)           3017           3062          65          5.2         191.8       0.8X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/DataSourceReadBenchmark-jdk11-results.txt:200]: Do you know the reason why this is slower than all ORC readers by at least 20%?

**Review by dongjoon-hyun — APPROVED** [review]

**Review by dongjoon-hyun — COMMENTED** [review]
One more question, is Parquet `DataPageV2` slower than Parquet `DataPageV1` in some cases?

File: sql/core/benchmarks/DataSourceReadBenchmark-jdk11-results.txt (hunk 1 above)
Comment (sunchao) [reply sql/core/benchmarks/DataSourceReadBenchmark-jdk11-results.txt:200]: I'm not sure. This part is calling the row-based Parquet reader from `parquet-mr` which I'm not very familiar with. Overall its performance is not very good.

**Comment by sunchao** [comment]
Thanks @viirya @dongjoon-hyun !

> One more question, is Parquet DataPageV2 slower than Parquet DataPageV1 in some cases?

Yes at the moment it is slower than V1 in several cases. I think @parthchandra has plan to improve its performance further.

**Comment by parthchandra** [comment]
> Thanks @viirya @dongjoon-hyun !
> 
> > One more question, is Parquet DataPageV2 slower than Parquet DataPageV1 in some cases?
> 
> Yes at the moment it is slower than V1 in several cases. I think @parthchandra has plan to improve its performance further.

I'll be submitting a couple of more PRs. Only one has the potential to improve performance. Let's see if it actually improves !

**Comment by sunchao** [comment]
Merged to master/3.3, thanks!

**Comment by LuciferYang** [comment]
Belated LGTM

**Review by c21 — COMMENTED** [review]
Late LGTM. We can add more benchmark later for deep nested type (e.g. struct of struct of struct of int).

**Pull request closed without merging**